zokrates_core/src/static_analysis/constant_resolver.rs

```diff
@@ -0,0 +1,844 @@
+// Static analysis step to replace all imported constants with the imported value
+// This does *not* reduce constants to their literal value
+// This step cannot fail as the imports were checked during semantics
+
+use crate::typed_absy::folder::*;
+use crate::typed_absy::*;
+use std::collections::HashMap;
+use zokrates_field::Field;
+
+// a map of the canonical constants in this program. with all imported constants reduced to their canonical value
+type ProgramConstants<'ast, T> =
+    HashMap<OwnedTypedModuleId, HashMap<ConstantIdentifier<'ast>, TypedConstant<'ast, T>>>;
+
+pub struct ConstantInliner<'ast, T> {
```

Comment:
should we call this as `ConstantResolver`? seems like the file name is different now